### ISSUE TITLE
Pagination action doesn't change page number

### ISSUE THREAD

**Issue by abishop451** [Sitecore/Sitecore.Commerce.Headstart.ReactAdmin#239]
On a Product List page (and likely other pagination functions throughout the app), when clicking the arrow pictured below, the page results go to the next page, but the page number displayed in the pagination box does not change. 

![image](https://github.com/Sitecore/Sitecore.Commerce.Headstart.ReactAdmin/assets/11871752/b9dc3767-71bc-4988-bc1d-72a893f72530)

**Comment by robertsoniv** [comment]
We have a new pattern for pagination that we are using on the SC/OC Portal - I will share the pattern on this task later today.

**Comment by robertsoniv** [comment]
I've had some success picking apart MUI pagination and rebuilding it using Chakra UI. Basically took the usePagination and useControlled hooks as is, then built out a Pagination component using the pagination hook and Chakra components.

Here are the docs for the usePagination hook that I took from MUI: https://mui.com/material-ui/react-pagination/#usepagination
 
I chose not install @mui/material/usePagination but that's definitely an option for the recipe - to make things a little more straight forward. We just didn't want to add another dependency to our project, so we pulled the code in manually.

You will also need to include useControlled as well.
